src/variableSubstitution.ts

```diff
@@ -99,7 +99,7 @@ const VARIABLE_SUBSTITUTIONS: VariableSubstitution[] = [
 export function getVariableSubstitutedAssociations(associations: XMLFileAssociation[]): XMLFileAssociation[] {
 
   // Collect properties needed to resolve variables
-  const currentFile: TextDocument = window.activeTextEditor.document;
+  const currentFile: TextDocument = !!window.activeTextEditor ? window.activeTextEditor.document: undefined;
```

Comment:
What about writing this code ?

```ts
const currentFile: TextDocument = window.activeTextEditor && window.activeTextEditor.document;
```

Reply:
I think we should check the languageId too:

`window.activeTextEditor.document.languageId`

Reply:
have you tried window?.activeTextEditor.document

Reply:
see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

Reply:
@Fred Doesn't work, I get a runtime error, but I'll check the langaugeId

Reply:
@datho7561 here the fix which is working for me:

```ts
const currentFile: TextDocument = window.activeTextEditor && window.activeTextEditor.document;
const currentFileUri: string = currentFile && currentFile.uri.fsPath;
  const currentWorkspace: WorkspaceFolder = currentFile && currentFile.uri && workspace.getWorkspaceFolder(currentFile.uri);
```